tests/integration-tests/conftest.py

```diff
@@ -1023,3 +1025,75 @@ def _build_image(image_id, region, cluster_config):
             run_command(["pcluster", "delete-image", "--id", image_id_post_test, "-r", region_post_test, "--force"])
             # sleep 90 seconds for image deletion
             time.sleep(90)
+
+
+@pytest.fixture(scope="class", autouse=True)
+def aws_user_credentials(region, scheduler, permissions_stack):
+    """
+    Define a fixture to create and assume the AWS credentials with the expected user policy.
+    :param region: region where the test is running
+    :param scheduler: scheduler used by the test
+    :param permissions_stack: the stack containing permissions resources
+    :return: a function to create and assume the AWS credentials with the expected user policy.
+    """
+
+    def get_role_arn(scheduler):
+        output_key = f"CustomerRole{scheduler.title()}"
+        return permissions_stack.cfn_outputs[output_key]
+
+    def assume_role(region, role_arn):
+        logging.info(f"Setting AWS credentials in region {region} for role: {role_arn}")
+        set_credentials(region, [f"{region},https://sts.{region}.amazonaws.com,{role_arn},jenkins"])
+        logging.info(f"Identity assumed in region {region}: {get_identity(region)}")
+
+    def reset_role():
+        logging.info("Restoring default AWS credentials")
+        unset_credentials()
+        logging.info(f"Identity restored in region {region}: {get_identity(region)}")
+
+    logging.info(f"Configuring AWS credentials in region {region} for scheduler: {scheduler}")
+
+    role_arn = get_role_arn(scheduler)
+    assume_role(region, role_arn)
+
+    yield
+
+    reset_role()
```

Comment:
I'm afraid this would break opt-in region tests cause it would leave no cred to be used by the tear down of other fixtures such as clusters_factory. 

This PR could address the problem but I did not have time to properly test it https://github.com/aws/aws-parallelcluster/pull/2777

Reply:
I think your approach should work at least in the scenario where we have only one level of set creds. 
If we want to support multiple levels, we should convert your backup dictionary to a backup stack of creds.

Reply:
Working on that in the next rev